### DISCUSSION TITLE
Trivy Scan fails for SDK 8.0 image (mcr.microsoft.com/dotnet/sdk:8.0)

### DISCUSSION THREAD

**Discussion by Ricky-G** [dotnet/dotnet-docker#5753] [Q&A]
Did a Trivy scan on mcr.microsoft.com/dotnet/sdk:8.0

and has the below issues, do we have to do this manually post pull of image or can the image be updated upstream for us please?

(nuget)
========
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-[38](https://github.com/.../platform/actions/runs/10234349836/job/28313721603#step:3:39)095 │ HIGH     │ fixed  │ 6.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/Containers/containerize/containerize.deps.json (dotnet-core)
=========================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/Containers/tasks/net472/Microsoft.NET.Build.Containers.deps.json (dotnet-core)
===========================================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                          Title                           │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────┤
│ Microsoft.IO.Redist │ CVE-2024-38081 │ HIGH     │ fixed  │ 6.0.0             │ 6.0.1         │ dotnet: Elevation of Privilege in VS Installer via NuGet │
│                     │                │          │        │                   │               │ config file                                              │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38081               │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/Containers/tasks/net8.0/Microsoft.NET.Build.Containers.deps.json (dotnet-core)
===========================================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/DotnetTools/dotnet-watch/8.0.303-servicing.2[43](https://github.com/.../platform/actions/runs/10234349836/job/28313721603#step:3:44)17.6/tools/net8.0/any/BuildHost-netcore/Microsoft.CodeAnalysis.Workspaces.MSBuild.BuildHost.deps.json (dotnet-core)
==============================================================================================================================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 7.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.[50](https://github.com/.../platform/actions/runs/10234349836/job/28313721603#step:3:51)9 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/DotnetTools/dotnet-watch/8.0.303-servicing.24317.6/tools/net8.0/any/dotnet-watch.deps.json (dotnet-core)
=====================================================================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/FSharp/fsc.deps.json (dotnet-core)
===============================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/FSharp/fsi.deps.json (dotnet-core)
===============================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/MSBuild.deps.json (dotnet-core)
============================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/NuGet.CommandLine.XPlat.deps.json (dotnet-core)
============================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/Roslyn/Microsoft.Build.Tasks.CodeAnalysis.deps.json (dotnet-core)
==============================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 7.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/dotnet/sdk/8.0.303/dotnet.deps.json (dotnet-core)
===========================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

usr/share/powershell/.store/powershell.linux.x[64](https://github.com/.../platform/actions/runs/10234349836/job/28313721603#step:3:65)/7.4.3/powershell.linux.x64/7.4.3/tools/net8.0/any/pwsh.deps.json (dotnet-core)
===============================================================================================================================
Total: 1 (HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 8.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘****

**Comment by richlander** [comment]
Can you share the commands you used to produce that result?

I used this: `docker run aquasec/trivy image --scanners vuln --severity MEDIUM,HIGH,CRITICAL mcr.microsoft.com/dotnet/sdk:8.0`

For NuGet, I get just this:

```bash
(nuget)
========
Total: 1 (MEDIUM: 0, HIGH: 1, CRITICAL: 0)

┌─────────────────────┬────────────────┬──────────┬────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Status │ Installed Version │ Fixed Version │                            Title                             │
├─────────────────────┼────────────────┼──────────┼────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ System.Formats.Asn1 │ CVE-2024-38095 │ HIGH     │ fixed  │ 6.0.0             │ 6.0.1, 8.0.1  │ dotnet: DoS when parsing X.509 Content and ObjectIdentifiers │
│                     │                │          │        │                   │               │ https://avd.aquasec.com/nvd/cve-2024-38095                   │
└─────────────────────┴────────────────┴──────────┴────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘
```

This issue is a false positive. It is the same as https://github.com/dotnet/dotnet-docker/issues/5325. We have been working to resolve all of these false positives. It appears that we still have work to do. Thanks for reporting.

**Comment by Ricky-G** [comment]
@richlander I am running Trivy as part of my GitHub pipeline and have set it to fail on any high/critical.  To get my pipeline to pass, looks like I will have to ignore this CVE-2024-38095 for now and later remove it once the false positive is removed.

Below is my command in the GitHub pipeline to produce that result

```
scan-sdk-image:
    runs-on: ubuntu-latest    
    steps:
    - name: Scan SDK base image with Trivy
      uses: aquasecurity/trivy-action@master
      with:
        image-ref: mcr.microsoft.com/dotnet/sdk:8.0
        format: 'table'
        exit-code: '1'
        ignore-unfixed: true
        vuln-type: 'os,library'
        severity: 'CRITICAL,HIGH'
```

Is it worth opening a new issue, or re-open #5325, would be good to have clean images, with out putting overrides or ignores :-)

**Reply by richlander** [reply]
I re-opened the issues with a bit more context. The `System.Formats.Asn1` is another false positive. We'll provide an update on when we have a fix for this.

**Reply by itaysk** [reply]
Hello from team Trivy :) Just chiming in to say that Trivy now allows software maintainers (you) to publish vulnerability analysis for your packages (or libraries or images) so that those vulnerabilities will be automatically suppressed for end users. You can see more info here:
https://aquasecurity.github.io/trivy/latest/docs/supply-chain/vex/repo/#publishing-vex-documents
https://github.com/aquasecurity/vexhub

**Reply by mthalman** [reply]
> Hello from team Trivy :) Just chiming in to say that Trivy now allows software maintainers (you) to publish vulnerability analysis for your packages (or libraries or images) so that those vulnerabilities will be automatically suppressed for end users. You can see more info here: https://aquasecurity.github.io/trivy/latest/docs/supply-chain/vex/repo/#publishing-vex-documents https://github.com/aquasecurity/vexhub

/cc @blowdart, @marcpopMSFT, @JonDouglas, @baronfel

**Reply by richlander** [reply]
That is good news. Thanks for sharing. We should explore that.

Some folks on our team are vex fans and have been advocating for this direction.

**Reply by itaysk** [reply]
cool. feel free to contact me/us if you have further questions or suggestions

**Reply by ericstj** [reply]
Which version of Trivvy is this?  https://github.com/aquasecurity/trivy/pull/7039 should improve this.

**Reply by itaysk** [reply]
vex-hub support is available in Trivy since v0.54 if that's what you're asking
